src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -83,9 +79,27 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
 
     @Override
     public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+        MatchAllQueryBuilder query = new MatchAllQueryBuilder();
+        query.fromXContent(parseContext);
+        return query.toQuery(parseContext);
+    }
+
+    public Query toQuery(QueryParseContext parseContext) {
+        if (boost == 1.0f) {
+            return Queries.newMatchAllQuery();
+        }
+
+        //LUCENE 4 UPGRADE norms field is not supported anymore need to find another way or drop the functionality
+        //MatchAllDocsQuery query = new MatchAllDocsQuery(normsField);
+        MatchAllDocsQuery query = new MatchAllDocsQuery();
```

Comment:
this comment is there from ages. Given that normsField doesn't get parsed for ages either, we should just drop any reference of it I guess. Wondering if we should do that upstream though or here. Just scared that we make too many changes at the same time and we don't track them. Thoughts?